packages/cli/src/util/dev/builder-cache.ts

```diff
@@ -1,412 +0,0 @@
-import npa from 'npm-package-arg';
-import pluralize from 'pluralize';
-import { basename, join } from 'path';
-import XDGAppPaths from 'xdg-app-paths';
-import { mkdirp, readJSON, writeJSON } from 'fs-extra';
-import { NowBuildError, PackageJson } from '@vercel/build-utils';
-import cliPkg from '../pkg';
-
-import cmd from '../output/cmd';
-import { Output } from '../output';
-import { NoBuilderCacheError } from '../errors-ts';
-
-import * as staticBuilder from './static-builder';
-import { BuilderWithPackage } from './types';
-import { isErrnoException } from '../is-error';
-
-const require_: typeof require = eval('require');
-
-const registryTypes = new Set(['version', 'tag', 'range']);
-
-const createStaticBuilder = (scope: string): BuilderWithPackage => {
-  return {
-    runInProcess: true,
-    requirePath: `${scope}/static`,
-    builder: Object.freeze(staticBuilder),
-    package: Object.freeze({ name: `@${scope}/static`, version: '' }),
-  };
-};
-
-const localBuilders: { [key: string]: BuilderWithPackage } = {
-  '@now/static': createStaticBuilder('now'),
-  '@vercel/static': createStaticBuilder('vercel'),
-};
-
-export const cacheDirPromise = prepareCacheDir();
-export const builderDirPromise = prepareBuilderDir();
-
-/**
- * Prepare cache directory for installing Vercel runtimes.
- */
-export async function prepareCacheDir() {
-  const designated = XDGAppPaths('com.vercel.cli').cache();
-
-  if (!designated) {
-    throw new NoBuilderCacheError();
-  }
-
-  const cacheDir = join(designated, 'dev');
-  await mkdirp(cacheDir);
-  return cacheDir;
-}
-
-export async function prepareBuilderDir() {
-  const builderDir = join(await cacheDirPromise, 'builders');
-  await mkdirp(builderDir);
-
-  // Create an empty `package.json` file, only if one does not already exist
-  try {
-    const buildersPkg = join(builderDir, 'package.json');
-    await writeJSON(buildersPkg, { private: true }, { flag: 'wx' });
-  } catch (err: unknown) {
-    if (!isErrnoException(err) || err.code !== 'EEXIST') {
-      throw err;
-    }
-  }
-
-  return builderDir;
-}
-
-function getNpmVersion(use = ''): string {
-  const parsed = npa(use);
-  if (registryTypes.has(parsed.type)) {
-    return parsed.fetchSpec || '';
-  }
-  return '';
-}
-
-export function getBuildUtils(packages: string[]): string {
-  const version = packages
-    .map(getNpmVersion)
-    .some(ver => ver.includes('canary'))
-    ? 'canary'
-    : 'latest';
-
-  return `@vercel/build-utils@${version}`;
-}
-
-function parseVersionSafe(rawSpec: string) {
-  try {
-    return semver.parse(rawSpec);
-  } catch (e) {
-    return null;
-  }
-}
-
-export function filterPackage(
-  builderSpec: string,
-  buildersPkg: PackageJson,
-  cliPkg: Partial<PackageJson>
-) {
-  if (builderSpec in localBuilders) return false;
-  const parsed = npa(builderSpec);
-  const parsedVersion = parseVersionSafe(parsed.rawSpec);
-
-  // Skip install of Runtimes that are part of Vercel CLI's `dependencies`
```

Comment:
I assume the answer is yes, but I just want to make sure: are we still skipping these for `vc build`?

Reply:
Yes, the `import-builders.test.ts` file tests for this explicitly.

Reply:
I'm looking at the tests and it doesn't seem to check if the "install" was skipped. Perhaps we need to check `client.stderr` for each one to ensure it wasn't installed.